src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -5,35 +5,34 @@
 #include <llprint.h>
 
 #define UNDEF_SYMBS 64
-
-/* Assembly function for sinv from new component */
-extern void *__inv_test_entry(int a, int b, int c);
+#define CMP_UTYPMEM_SZ (1 << 26)
+#define THD_PRIO 5
+#define THD_PERIOD 10
+#define THD_BUDGET 5
 
 struct cobj_header *hs[MAX_NUM_SPDS + 1];
 
 /* The booter uses this to keep track of each comp */
 struct comp_cap_info {
-	struct cos_compinfo *compinfo;
-	struct usr_inv_cap   ST_user_caps[UNDEF_SYMBS];
-	vaddr_t              vaddr_user_caps; // vaddr of user caps table in comp
-	vaddr_t              addr_start;
-	vaddr_t              vaddr_mapped_in_booter;
-	vaddr_t              upcall_entry;
+	struct cos_compinfo    *compinfo;
+	struct cos_defcompinfo *defcompinfo;
+	struct usr_inv_cap      ST_user_caps[UNDEF_SYMBS];
+	vaddr_t                 vaddr_user_caps; // vaddr of user caps table in comp
+	vaddr_t                 addr_start;
+	vaddr_t                 vaddr_mapped_in_booter;
+	vaddr_t                 upcall_entry;
 } new_comp_cap_info[MAX_NUM_SPDS + 1];
 
-struct cos_compinfo boot_info;
-struct cos_compinfo new_compinfo[MAX_NUM_SPDS + 1];
-
-int                      schedule[MAX_NUM_SPDS + 1];
-volatile size_t          sched_cur;
+struct cos_compinfo *boot_info;
+struct cos_defcompinfo  new_defcompinfo[MAX_NUM_SPDS + 1];
```

Comment:
Can we not inline this into `struct comp_cap_info`? And have wrappers to get compinfo or defcompinfo.